### ISSUE TITLE
apr run produces garbage output on Intel Xeon CPU (PPL 47,350)

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#617]
## Description

`apr run` generates completely incoherent output on Intel Xeon W-3245 (x86_64, no GPU, CPU-only inference). The same model produces correct output on lambda-labs (RTX 4090 CUDA) and correct throughput on `apr bench` on the same Xeon.

## Reproduction

```bash
# On Intel Xeon W-3245 (CPU only)
$ apr run model.gguf "What is 2+2?" --max-tokens 16
Output: |് ه 发 ه 主 冗？？？？？？

$ apr eval model.gguf --max-tokens 64
Perplexity: 47350.89 (FAIL: > 20.0)
Quality: Garbage (model broken)
```

Compare on lambda-labs (RTX 4090):
```bash
$ apr run model.gguf "What is 2+2?" --max-tokens 16
Output: 2+2 is 4.

$ apr eval model.gguf --max-tokens 64
Perplexity: 18.94 (PASS)
```

Meanwhile `apr bench` on the same Xeon reports 25.7 tok/s and runs successfully — suggesting the bench path uses a different (working) inference code path than `run`.

## Impact

**P0** — CPU inference is broken on Intel Xeon. Output is random Unicode garbage. This makes `apr run` unusable on CPU-only x86_64 servers.

## Five Whys Root Cause Analysis

1. **Why garbage?** The generated tokens are random/wrong — the inference forward pass produces incorrect logits on CPU.
2. **Why incorrect logits?** `apr bench` works correctly on the same CPU — different code path. `run` likely takes a wgpu path that falls back incorrectly, while `bench` uses the realizar CPU path directly.
3. **Why different code paths?** `apr run` attempts GPU init (wgpu) first and may corrupt state before falling back to CPU. `apr bench` explicitly checks "CUDA: Not available (CPU mode)" and skips GPU entirely.
4. **Why state corruption?** The wgpu shader compilation fails (`-inf` bug, #598) and the fallback to CPU may use partially initialized GPU tensors or wrong memory layouts.
5. **Why no contract enforcement?** No `layer-parity-v1.yaml` contract check runs during `apr run`. If `apr parity` were integrated as a pre-flight check, it would catch CPU/GPU divergence before returning results to users.

## Root Cause

Missing provable-contracts enforcement. The `layer-parity-v1.yaml` contract exists but is only checked by `apr parity` (which itself has exit-code bugs #601/#615). It is not enforced in the `apr run` hot path. A failed GPU init corrupts the inference state, and no contract gate prevents garbage from reaching the user.

## Version

`apr 0.4.13` on Intel Xeon W-3245
